### PR TITLE
add common module

### LINKED ISSUE #486: introduce "common" module for code shared between modules
- [ ] Add a new maven module "common" under "modules".
- [ ] This module should have zero dependency to other modules so that it can be easily imported into other modules without having circular- dependencies.
- [ ] Move shared methods from OpenApiUtil to this common module with the same class name - OpenApiUtil. 
- [ ] Move the service API methods from OpenApiUtil to "register" package with class name RegisteredServicesOpenApiUtil
- [ ] Use classes from common/OpenApiUtil in RegisteredServicesOpenApiUtil and CredentialsOpenApiUtil

### PR DESCRIPTION
**Pull request by swaroopar**
fixes #486

### CODE REVIEW

File: modules/common/src/main/java/org/eclipse/xpanse/common/openapi/OpenApiUtil.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                URLConnection con = url.openConnection();
+                ReadableByteChannel rbc = Channels.newChannel(con.getInputStream());
+                fos = new FileOutputStream(downloadTemp);
+                fos.getChannel().transferFrom(rbc, 0, Long.MAX_VALUE);
+                downloadEnd = true;
+                log.info("Download openapi client:{} from URL:{} end.",
+                        execJarFile.getPath(), clientDownLoadUrl);
+            } finally {
+                if (Objects.nonNull(fos)) {
+                    fos.close();
```
[the remark this answers is not in the document]
Comment (swaroopar) [reply modules/common/src/main/java/org/eclipse/xpanse/common/openapi/OpenApiUtil.java:119]: @sonatype-lift ignore

File: modules/register/src/main/java/org/eclipse/xpanse/modules/register/register/utils/RegisteredServicesOpenApiGenerator.java [hunk 2]
```diff
@@ -212,7 +134,7 @@ public String createServiceApi(RegisterServiceEntity registerService) {
```
[the remark this answers is not in the document]
Comment (swaroopar) [reply modules/register/src/main/java/org/eclipse/xpanse/modules/register/register/utils/RegisteredServicesOpenApiGenerator.java]: @sonatype-lift ignore

File: modules/common/src/main/java/org/eclipse/xpanse/common/openapi/OpenApiUtil.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+    /**
+     * Get the work directory of the openApi.
+     *
+     * @return workdir  The work directory of the openApi.
+     */
+    public String getOpenApiWorkdir() {
+        String rootPath = System.getProperty("user.dir");
+        try {
+            int modulesIndex = rootPath.indexOf("modules");
```
[the remark this answers is not in the document]
Comment (swaroopar) [reply modules/common/src/main/java/org/eclipse/xpanse/common/openapi/OpenApiUtil.java:140]: @sonatype-lift ignore

File: modules/common/src/main/java/org/eclipse/xpanse/common/openapi/OpenApiUtil.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        if (!execJarFile.exists() && !execJarFile.canExecute()) {
+            log.info("Download openapi client:{} from URL:{} start.",
+                    execJarFile.getPath(), clientDownLoadUrl);
+            String jarTempFile = execJarName + ".temp";
+            File downloadTemp = new File(workDir, jarTempFile);
+            FileOutputStream fos = null;
+            boolean downloadEnd;
+            try {
+                URL url = new URL(clientDownLoadUrl);
+                URLConnection con = url.openConnection();
```
[the remark this answers is not in the document]
Comment (swaroopar) [reply modules/common/src/main/java/org/eclipse/xpanse/common/openapi/OpenApiUtil.java:110]: @sonatype-lift ignore

File: modules/register/src/main/java/org/eclipse/xpanse/modules/register/register/utils/RegisteredServicesOpenApiGenerator.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (swaroopar) [reply modules/register/src/main/java/org/eclipse/xpanse/modules/register/register/utils/RegisteredServicesOpenApiGenerator.java]: @sonatype-lift ignore

File: modules/register/src/main/java/org/eclipse/xpanse/modules/register/register/utils/RegisteredServicesOpenApiGenerator.java [hunk 5]
```diff
@@ -231,14 +153,13 @@ public String createServiceApi(RegisterServiceEntity registerService) {
```
[the remark this answers is not in the document]
Comment (swaroopar) [reply modules/register/src/main/java/org/eclipse/xpanse/modules/register/register/utils/RegisteredServicesOpenApiGenerator.java]: @sonatype-lift ignore

**Review by iskey — APPROVED** [review]
looks good.

**Pull request merged**